pkg/services/accesscontrol/middleware/middleware.go

```diff
@@ -38,15 +42,47 @@ func Middleware(ac accesscontrol.AccessControl) func(macaron.Handler, string, ..
 
 			hasAccess, err := ac.Evaluate(c.Req.Context(), c.SignedInUser, permission, runtimeScope...)
 			if err != nil {
-				c.Logger.Error("Error from access control system", "error", err)
-				c.JsonApiErr(http.StatusForbidden, "Forbidden", nil)
+				deny(c, permission, runtimeScope, err)
 				return
 			}
 			if !hasAccess {
-				c.Logger.Info("Access denied", "userID", c.UserId, "permission", permission, "scopes", runtimeScope)
-				c.JsonApiErr(http.StatusForbidden, "Forbidden", nil)
+				deny(c, permission, runtimeScope, nil)
 				return
 			}
 		}
 	}
 }
+
+func deny(c *models.ReqContext, permission string, scopes []string, err error) {
```

Comment:
Does it make sense to make this public so that we can use it in the [enterprise](https://github.com/grafana/grafana-enterprise/blob/master/src/pkg/extensions/accesscontrol/api/evaluator.go#L11)?